cpp/cmake/deps/nvtx3.cmake

```diff
@@ -0,0 +1,40 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+if (NOT TARGET deps::nvtx3)
+    FetchContent_Declare(
+            deps-nvtx3
+            GIT_REPOSITORY https://github.com/NVIDIA/NVTX.git
+            GIT_TAG 3c98c8425b0376fd8653aac7cfc6a864f3897752
+            # GIT_SHALLOW TRUE  # Uncomment this when the official release of nvtx3-cpp is available.
```

Comment:
Should we raise an issue to track this?

Reply:
Thanks @jakirkham!

Created #168 and updated the comment in `nvtx3.cmake`.